### PR TITLE
Xray usage event report

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Adding back Xray event reports to be sent as default.
The issue that only admins could submit those events was fixed at `v3.83` this was enforced at:
https://github.com/jfrog/jfrog-client-go/pull/833

### CODE REVIEW

File: utils/usage/usage.go [hunk 1]
```diff
@@ -89,6 +93,14 @@ func (ur *UsageReporter) Report(features ...ReportFeature) {
 			return
 		})
 	}
+	if ur.sendToXray {
+		ur.reportWaitGroup.Go(func() (err error) {
+			if err = ur.reportToXray(features...); err != nil {
+				err = fmt.Errorf("xray, %s", err.Error())
```
Comment (yahavi) [inline utils/usage/usage.go:99]: Let's do it in "ecosystem" and "artifactory" too
```suggestion
				err = fmt.Errorf("xray, %w", err)
```

File: utils/usage/usage.go [hunk 2]
```diff
@@ -122,6 +134,23 @@ func (ur *UsageReporter) reportToEcosystem(features ...ReportFeature) (err error
...
+	}
+	serviceManager, err := xrayutils.CreateXrayServiceManager(ur.serverDetails)
+	if err != nil {
+		return
+	}
+	events := ur.convertAttributesToXrayEvents(features...)
+	if len(events) == 0 {
+		err = errorutils.CheckErrorf("Nothing to send.")
+		return
+	}
```
Comment (yahavi) [inline utils/usage/usage.go:150]: Can we run `if len(features) == 0` check before reporting to "artifactory", "xray", and "ecosystem"?

File: utils/usage/usage_test.go [hunk 3]
```diff
@@ -129,6 +131,64 @@ func createArtifactoryUsageHandler(t *testing.T, productName, commandName string
...
+		ClientId:  clientName,
+	})
+	assert.NoError(t, reporter.WaitForResponses())
+}
+
+func TestReportXrayError(t *testing.T) {
+	reporter := NewUsageReporter("", &config.ServerDetails{}).SetSendToEcosystem(false).SetSendToArtifactory(false)
+	reporter.Report(ReportFeature{
+		FeatureId: "",
+	})
```
Comment (yahavi) [inline utils/usage/usage_test.go:156]:
```suggestion
	reporter.Report(ReportFeature{})
```

File: utils/usage/usage_test.go [hunk 4]
```diff
@@ -129,6 +131,64 @@ func createArtifactoryUsageHandler(t *testing.T, productName, commandName string
...
+	assert.Error(t, reporter.WaitForResponses())
+
+	server := httptest.NewServer(create404UsageHandler(t))
+	defer server.Close()
+	reporter = NewUsageReporter("", &config.ServerDetails{ArtifactoryUrl: server.URL + "/"}).SetSendToEcosystem(false).SetSendToArtifactory(false)
+	reporter.Report(ReportFeature{
+		FeatureId: "",
+	})
```
Comment (yahavi) [inline utils/usage/usage_test.go:164]:
```suggestion
	reporter.Report(ReportFeature{})
```

File: utils/usage/usage_test.go [hunk 5]
```diff
@@ -129,6 +131,64 @@ func createArtifactoryUsageHandler(t *testing.T, productName, commandName string
...
+			_, err := buf.ReadFrom(r.Body)
+			assert.NoError(t, err)
+			featureId := xrayusage.GetExpectedXrayEventName(productId, commandName)
+			assert.Equal(t, fmt.Sprintf(`[{"data":{"clientId":"%s"},"product_name":"%s","event_name":"%s","origin":"API_CLI"}]`, clientId, productId, featureId), buf.String())
+
+			// Send response OK
+			w.WriteHeader(http.StatusOK)
+			_, err = w.Write([]byte("{}"))
+			assert.NoError(t, err)
+		}
```
Comment (yahavi) [inline utils/usage/usage_test.go:188]: Let's check that there are no unexpected requests
```suggestion
			return
		}
		assert.Fail(t, "Unexpected request URI", r.RequestURI)
```

**Review by yahavi — CHANGES_REQUESTED** [review]

File: utils/usage/usage.go (hunk 2 above)
Comment (attiasas) [reply utils/usage/usage.go]: each type of report has different attributes and limitations, even if features > 0 there could be an option for some converted format to be empty, added a general check before reporting and moved this check for each report to be the first thing

**Pull request merged**